ot/unbalanced/_sinkhorn.py

```diff
@@ -865,6 +895,252 @@ def sinkhorn_stabilized_unbalanced(a, b, M, reg, reg_m, reg_type="kl", c=None,
+    -------
+    if n_hists == 1:
+        - gamma : (dim_a, dim_b) array-like
+            Optimal transportation matrix for the given parameters
+        - log : dict
+            log dictionary returned only if `log` is `True`
+    else:
+        - ot_distance : (n_hists,) array-like
+            the OT distance between :math:`\mathbf{a}` and each of the histograms :math:`\mathbf{b}_i`
+        - log : dict
+            log dictionary returned only if `log` is `True`
+
+    Examples
+    --------
+
+    >>> import ot
+    >>> a=[.5, .5]
+    >>> b=[.5, .5]
+    >>> M=[[0., 1.],[1., 0.]]
+    >>> ot.unbalanced.sinkhorn_unbalanced_translation_invariant(a, b, M, 1., 1.)
+    array([[0.32205357, 0.11847689],
+           [0.11847689, 0.32205357]])
+
+    .. _references-sinkhorn-unbalanced-translation-invariant:
+    References
+    ----------
+    .. [70] Séjourné, T., Vialard, F. X., & Peyré, G. (2022).
+       Faster unbalanced optimal transport: Translation invariant sinkhorn and 1-d frank-wolfe.
+       In International Conference on Artificial Intelligence and Statistics (pp. 4995-5021). PMLR.
+    """
+
+    M, a, b = list_to_array(M, a, b)
+    nx = get_backend(M, a, b)
+
+    dim_a, dim_b = M.shape
+
+    if len(a) == 0:
+        a = nx.ones(dim_a, type_as=M) / dim_a
+    if len(b) == 0:
+        b = nx.ones(dim_b, type_as=M) / dim_b
+
+    if len(b.shape) > 1:
+        n_hists = b.shape[1]
+    else:
+        n_hists = 0
+
+    reg_m1, reg_m2 = get_parameter_pair(reg_m)
+
+    if log:
+        dict_log = {'err': []}
+
+    # we assume that no distances are null except those of the diagonal of
+    # distances
+    if warmstart is None:
+        if n_hists:
+            u = nx.ones((dim_a, 1), type_as=M)
+            v = nx.ones((dim_b, n_hists), type_as=M)
+            a = a.reshape(dim_a, 1)
+        else:
+            u = nx.ones(dim_a, type_as=M)
+            v = nx.ones(dim_b, type_as=M)
+    else:
+        u, v = nx.exp(warmstart[0]), nx.exp(warmstart[1])
+
+    u_, v_ = u, v
+
+    if reg_type == "entropy":
+        warnings.warn('If reg_type = entropy, then the matrix c is overwritten by the one matrix.')
+        c = nx.ones((dim_a, dim_b), type_as=M)
+
+    if n_hists:
+        M0 = M
+    else:
+        c = a[:, None] * b[None, :] if c is None else c
+        M0 = M - reg * nx.log(c)
+    K = nx.exp(-M0 / reg)
+
+    fi_1 = reg_m1 / (reg_m1 + reg) if reg_m1 != float("inf") else 1
+    fi_2 = reg_m2 / (reg_m2 + reg) if reg_m2 != float("inf") else 1
+
+    k1 = reg * reg_m1 / ((reg + reg_m1) * (reg_m1 + reg_m2)) if reg_m1 != float("inf") else 0
+    k2 = reg * reg_m2 / ((reg + reg_m2) * (reg_m1 + reg_m2)) if reg_m2 != float("inf") else 0
+
+    k_rho1 = k1 * reg_m1 / reg if reg_m1 != float("inf") else 0
+    k_rho2 = k2 * reg_m2 / reg if reg_m2 != float("inf") else 0
+
+    if reg_m1 == float("inf") and reg_m2 == float("inf"):
+        xi1, xi2 = 0, 0
+        fi_12 = 1
+    elif reg_m1 == float("inf"):
+        xi1 = 0
```

Comment:
Why are these line sot tested? imaybe the test shoudl run with infinite values too

Reply:
I think the test only handled ("inf", "inf"). I added a test to handle the semi-relaxed case.